build.rs

```diff
@@ -20,10 +20,24 @@ fn main() {
         None => return,
     };
 
-    // If the target isn't thumbv6 then we can use atomic CAS
-    if !target.starts_with("thumbv6") {
-        println!("cargo:rustc-cfg=atomic_cas");
-    }
+    match &target[..] {
```

Comment:
Maybe we could extract these into little functions so they're a bit easier to follow:

```rust
if target_has_atomic_cas { .. }

..

if target_has_atomics() { .. }
```